jupyter_rsession_proxy/__init__.py

```diff
@@ -29,13 +32,46 @@ def setup_rserver():
     def _get_env(port):
         return dict(USER=getpass.getuser())
 
+    def db_config():
+        '''Create a temporary directory to hold rserver's database.'''
+        db_dir = tempfile.TemporaryDirectory()
+        # create the rserver database config
+        db_conf = dedent("""
+            provider=sqlite
+            directory={directory}
+        """).format(directory=db_dir.name)
+        f = tempfile.NamedTemporaryFile(mode='w', delete=False)
+        db_config_name = f.name
+        f.write(db_conf)
+        f.close()
+        return db_config_name
+
     def _get_cmd(port):
-        return [
+        cmd = [
             get_rstudio_executable('rserver'),
+            '--auth-none=1',
+            '--www-frame-origin=same',
             '--www-port=' + str(port),
-            '--www-frame-origin=same'
+            '--www-verify-user-agent=0'
         ]
 
+        if os.environ.get('RSESSION_PROXY_DB_CONFIG', '0') != '0':
+            cmd.append(f'--database-config-file={db_config()}')
+
+        # Tell rserver what path it is being served from.
+        # rserver's www-root-path option is present in RStudio >= 1.4. If this
+        # environment variable is set to rserver's default, /, we don't set the
+        # option. The value should be the name of our entry point which is
+        # usually /rstudio/, however since people can add custom entry points,
+        # we won't assume.
+        www_root_path = os.environ.get('RSESSION_PROXY_WWW_ROOT_PATH', '/')
+        if www_root_path != "/":
+            hub_service_prefix = os.environ.get('JUPYTERHUB_SERVICE_PREFIX', '')
+            root_path = pathlib.Path(hub_service_prefix + www_root_path)
+            cmd.append(f'--www-root-path={root_path}/')
```

Comment:
If you instead use `--www-root-path={base_url}`, it should automatically set the appropriate base url.

Reply:
I forgot about that, thanks! Though I think it needs to be `{base_url}/rstudio/`.